draft-birkholz-scitt-receipts.md

```diff
@@ -294,35 +299,36 @@ ProofElement = [
 
 LeafInfo = [
     internal_hash: bstr,
-    internal_data: bstr,
-    sign_protected: empty_or_serialized_map
+    internal_data: bstr
 ]
 ~~~
 
 ## Receipt Verification
 
-Given the TS parameters, a signed envelope, and a Receipt for it,
+Given a signed envelope and a Receipt for it,
 the following steps must be followed to verify this Receipt.
 
-1. Verify that the Receipt Content structure is well-formed, as described in {{ReceiptContents}}.
+1. Decode the protected header of the Receipt and look-up the TS parameters using the service_id field.
+
+2. Verify that the Receipt Content structure is well-formed, as described in {{ReceiptContents}}.
 
-2. Construct a `Countersign_structure` as described in {{cose_sign1_countersign}}, using `sign_protected` from the `leaf_info` field of the receipt contents.
+3. Construct a `Countersign_structure` as described in {{cose_sign1_countersign}}, using the protected header of the Receipt as `sign_protected`.
 
-3. Compute `LeafBytes` as the bytestring concatenation of the internal hash, the hash of internal data, and the hash of the CBOR-encoding of `Countersign_structure`, using the CBOR encoding described in {{deterministic-cbor}}.
+4. Compute `LeafBytes` as the bytestring concatenation of the internal hash, the hash of internal data, and the hash of the CBOR-encoding of `Countersign_structure`, using the CBOR encoding described in {{deterministic-cbor}}.
 
         LeafBytes := internal_hash || HASH(internal_data) || HASH(cbor(Countersign_structure))
 
-4. Compute the leaf digest.
+5. Compute the leaf digest.
```

Comment:
You might want to salt before computing the leaf to avoid:

https://github.com/w3c-ccg/Merkle-Disclosure-2021/issues/4
https://github.com/w3c-ccg/Merkle-Disclosure-2021/issues/3

It introduces additional complexity though...

Reply:
In this case it's not an issue, as leaves here are always the hash of a concatenation of three hashes whereas intermediate nodes are the hash of two hashes, so they can never be confused and lead to such attacks.